### PR TITLE
Profile Changer

### PR DESCRIPTION
**Pull request by caiocinel**
Warning: This is a Pull Request without any merge interest, it only serves as a basis for someone who wants to add a ProfileChanger to their build.

The hook method is vfunc, it needs adaptation for the new libraries.

The added files already have full capacity for adding an InventoryChanger / MedalChanger, but I didn't have the patience for that.

If you know how to optimize or extend, feel free to share, that I will try to do the best (or make a pull request on the fork).

In short: it has what it takes to function and nothing else. 

(se você for Brasileiro: Isso é uma pura gambiarra).

Credits: a set of several leaked sources that I found in corsair.wtf;

### CODE REVIEW

**Comment by TarwzDev** [comment]
Aprendeu omigo

File: Osiris/Changer/valve_parser.h [hunk 1]
```diff
@@ -0,0 +1,947 @@
...
+			root = std::make_shared<Node>(this);
+			return root->Parse();
+		}
+	};
+}
+
+
+
+
+static const char smalll_compressed_data_base85[32600 + 1] =
```
Comment (linuxer9) [inline Osiris/Changer/valve_parser.h]: yo , remove that junk!

**Review by linuxer9 — CHANGES_REQUESTED** [review]

**Comment by doggobot-owner** [comment]
You need to add a config system, it doesnt save

**Comment by Scui1** [comment]
can't change ranks on wingman and dangerzone, nice paste 😂

**Comment by caiocinel** [comment]
Be a little less ignorant and read the description of the pull request.

Em qui., 21 de jan. de 2021 às 10:32, Scui1 <notifications@github.com>
escreveu:

**Comment by doggobot-owner** [comment]
this should be developed more and added medal and inventory changer, ngl
also add config system
goodluck

**Comment by thisisnotagithub** [comment]
works as it should thank you for your contribution. i will continue to try and develop this project and add a medal changer, i would appreciate it if other people tried as well

**Comment by MirzaGorgAli** [comment]
Hi and thank you @caiocinel for this awesome pull. Can anyone make medal changer that other people be able to see medals on my profile in lobby (not in the match only when they join my lobby)?
*Sorry about my ENGLISH*

**Comment by linuxer9** [comment]
I'm working on the inventory changer, I finally managed to make it work but I have "a stupid" issue, I can't get steam account ID 
`uint32_t steamid = memory->SteamUser->GetSteamID().GetAccountID();`
I tried adding this code to memory.cpp
`ISteamUser* SteamUser = SteamClient->GetISteamUser((void*)1, (void*)1, "SteamUser019");`
but the game is crashing..
I manually hardcoded my steam ID and it works fine, 
@caiocinel

**Comment by caiocinel** [comment]
> I'm working on the inventory changer, I finally managed to make it work but I have "a stupid" issue, I can't get steam account ID
> I manually hardcoded my steam ID and it works fine,
> @caiocinel

Take a look at how Daniel does it on GOESP:

https://github.com/danielkrupinski/GOESP/search?q=steamid

**Comment by linuxer9** [comment]
> 
> Take a look at how Daniel does it on GOESP:
> 
> https://github.com/danielkrupinski/GOESP/search?q=steamid

Thanks a took a look at it before and it's not what I want, It grabs steam IDs of entities during the game, so it's not helpful if the user is in the main menu (i.e. LocalPlayer doesn't exist)

**Comment by linuxer9** [comment]
I feel dumb, I set the steam id to 0 and it still works
```
void Inventory::ApplyMedals(ProtoWriter& object)
{
	uint32_t MedalIndex = 970;// config->medalChanger.medals;
	uint64_t steamid = 0;
...
```
do you think this might have side effects?

**Comment by Scui1** [comment]
> 
> 
> I'm working on the inventory changer, I finally managed to make it work but I have "a stupid" issue, I can't get steam account ID
> `uint32_t steamid = memory->SteamUser->GetSteamID().GetAccountID();`
> I tried adding this code to memory.cpp
> `ISteamUser* SteamUser = SteamClient->GetISteamUser((void*)1, (void*)1, "SteamUser019");`
> but the game is crashing..
> I manually hardcoded my steam ID and it works fine,
> @caiocinel

wtf are you doing? Why are you passing 1 as values for handles? maybe read the [documentation](https://partner.steamgames.com/doc/api/ISteamClient)

**Comment by MirzaGorgAli** [comment]
I'm noob but maybe :
"ISteamUser* SteamUser;" in the "memory.h"
"SteamUser = (ISteamUser*)SteamClient->GetISteamUser((void*)1, (void*)1, "SteamUser019");" in the "memory.cpp"

Test an tell me :D

**Comment by caiocinel** [comment]
Memory.cpp:
` SteamUser = SteamClient->GetISteamUser((void*)1, (void*)1, "SteamUser019"
);  `

Memory.h
` ISteamUser* SteamUser;  `

Call with:
memory->SteamUser->GetSteamID().GetAccountID();

Em sáb., 13 de fev. de 2021 às 02:40, MirzaGorgAli <notifications@github.com>
escreveu:

**Comment by linuxer9** [comment]
> 
> wtf are you doing? Why are you passing 1 as values for handles? maybe read the [documentation](https://partner.steamgames.com/doc/api/ISteamClient)

I have no idea of I what I'm doing, actually but I looked at "GetISteamGenericInterface" it's the function and he passed 1 for both hSteamUser and hSteamPipe, so I did the same. I appreciate your tip :)

I finally solved it, thanks to @MirzaGorgAli and @caiocinel, your both solutions did actually work :) 👍

**Comment by zioloprime** [comment]
How to make usage of this/?

**Comment by FlameProfi** [comment]
autoupdate skins don't work, please fix

**Comment by caiocinel** [comment]
> autoupdate skins don't work, please fix

See what causes the problem and do a review, with that I can fix the problem.

This project is an example, it is far from ideal.

**Comment by linuxer9** [comment]
> 
> 
> autoupdate skins don't work, please fix

changing local inventory doesn't change your the server side inventory.
the game still gets your items from GC server.
so in order to make inventory changer work in game, you have two options:
- either force fallback paintkit on weapons according to the equipped weapon slot.
- or, dynamically get the item ID of equipped slot and set your current weapon item id to that id from local inventory.
you still need to change models, fix animations of knives and make gloves, just like the current skin changer.

**Comment by caiocinel** [comment]
I'm working on a better ProfileChanger, allowing me to change Wingman and DZ as well.

**Comment by SquidmarcHD** [comment]
hey can i get your discord by any chance?

**Comment by caiocinel** [comment]
Caillou#9999

Em seg., 24 de mai. de 2021 às 08:32, SquidmarcHD ***@***.***>
escreveu:

**Comment by SquidmarcHD** [comment]
would be cool if you also accept it :+1:

**Comment by caiocinel** [comment]
Unfortunately this week I'm busy, maybe on Saturday.

Em ter., 25 de mai. de 2021 às 14:20, SquidmarcHD ***@***.***>
escreveu:

**Comment by alerion921** [comment]
Will this be updated?

**Comment by caiocinel** [comment]
probably not

Em qua., 13 de jul. de 2022 às 17:27, alerion921 ***@***.***>
escreveu:

**Comment by buttonzero** [comment]
does this work still, been lookin for a profile changer for a while cant seem to find one
